numpy/_core/multiarray.py

```diff
@@ -547,11 +583,64 @@ def lexsort(keys, axis=None):
     [1 0 3 2]
 
     """
-    if isinstance(keys, tuple):
-        return keys
-    else:
-        return (keys,)
 
+    import numpy as np
+
+    if np.ndim(keys[0]) == 0:
+        # Weird case, each level is actually a scalar, to be interpreted as a
+        # level of length 1:
+        if axis:
+            msg = f"axis {axis} is out of bounds for array of dimension 0"
+            raise np.exceptions.AxisError(msg)
+        return 0
+
+    if len(keys[0]) > 1000:
+        n_layers = len(keys)
+        # Check if we are only dealing with integers:
+        if all([isinstance(a, np.ndarray) and
+                np.issubdtype(a.dtype, np.integer)
+                for a in keys]):
```

Comment:
Is there a tab here? The formatting is strange

Reply:
@mattip Outdated, but for the records: no, I just had aligned the content of the square brackets... something similar happens in the new version at https://github.com/numpy/numpy/pull/26434/files#diff-33968b6e8acadeaac162e52ce132395af75712779b23ad0e3cb1600fb36edcc6R659